### ISSUE TITLE
How is LRPZ same as input_t_gradient? Citation for LRPZ requested.

### ISSUE THREAD

**Issue by vedhas** [albermax/innvestigate#126]
* In your talk https://www.youtube.com/watch?v=LtbM2phNI7I (13:25, 37:28, 1:23:50), you have emphasised that LRP is different from input*gradient. I also understand that there are different flavours of LRP. But isn't it true that LRP-Z is same likely same as input times gradient? I found this to be the case when running my experiments and finding the results for the two analyzers to be identical, which made me curious. 
(When digging it down further, I found that indeed the [documentation for LRP-Z](https://github.com/albermax/innvestigate/blob/master/innvestigate/analyzer/relevance_based/relevance_analyzer.py)  (class LRPZ/BaselineLRPZ) confirms this. The comment reads: `Applies the "LRP-Z" algorithm to analyze the model. Based on the gradient times the input formula.`) 

* Kindly let me know how exactly LRP-Z relates to other LRP versions, i.e. if it has different hyperparameters? Kindly direct me to the paper mentioning LRP-Z specifically, as I could not find one (with that name at least). 

I am attempting a regression problem, and I hope I am on the right track looking at the model visualisation. I need to cite innvestigate and LRP-Z in the next few days if my experiments get successful. Kindly reply soon! Thank you very much in advance :) :+1:

**Comment by albermax** [comment]
Hi Vedhas!

LRP-Z is only equal to input\*gradient if the network is composed of linear, ReLU and max-pooling layers etc, otherwise it is different!
Thank you for pointing that out. I will update the documentation. FYI: That's the reason this analyzer can only be called with a reduced set of layers.

LRP-Z was introduced in the first paper: [PlosONE](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0130140)
To cite iNNvestigate please use the arXiv version in the readme. Thanks!

@sebastian-lapuschkin Could you please answer the question how LRP-Z relates to other versions of LRP?

**Comment by sebastian-lapuschkin** [comment]
Hi vedhas, 

LRP-Z can be regarded as the basic building block of LRP, ie it implements conservative decomposition of upstream relevance in proportion to quantities z_{ij} sent from downstream neurons i to upstream neurons j in the forward pass.

While z_{ij} describes in principle any mapping possible, for DNNs it is the linear projection x_i*w_{ij} for layers with weight kernels.

Then, there are further purposed rules such as the epsilon decomposition, which adds a small numerical stabilizer to the decomposition's denominator for avoiding divisions by zero (in case mappings z_{ij} cancel each other out) and thus reduce overly amplified relevance peaks.

The alpha-beta decomposition treats positive and negative forward contributions separately, which matches well with DNNs using ReLU activation units. The special case of alpha=1, beta=0 is used as "ExcitationBackpropagation", for example.

Other rules are for example the "flat" rule, which in the backward pass overwrites all z_{ij} with the value 1. This is equivalent to a uniform projection of an upstream neuron's relevance to that neuron's receptive field.
Intuitively this results in a lower granularity of relevance maps representing a  hidden embedding of the network input (and does not result in meaningful information for fully connected layers).

For DNNs, LRP can be formulated as a modified gradient backward pass.
The main difference of LRP and Gradient times Input is the treatment of activation functions, as max already has said.
The end result is that LRP can backpropagate through activation layers with saturated activations:
Consider the tanh activation with a very high or low input. Here, the gradient would be zero and nothing is backpropagated. LRP however does not compute the gradient but considers the (proportional) activation of the neuron and thus can backpropagate relevance.
